### ISSUE TITLE
v5.0.0-beta3 extra cells cause validation problems

### ISSUE THREAD

**Issue by 200Puls** [vaadin/vaadin-grid#1241]
### Description
I'm using vaadin-grid together with iron-form to have editable rows and validation. This setup worked perfectly on v3.

Unfortunately vaadin-grid v5.0.0-beta3 renders additional instances of my components which are not visible but taken into account when validation occurs.

### Expected outcome
I would expect that only as many column elements are rendered as there are entries in the items array.

### Actual outcome
Additonal element instances are created in `_columnTreeChanged`  and `_updateRow` in vaadin-grid.
If I have one row in the items array and just one column with paper-input 3 instances of paper-input are created and validated.

### Live Demo
https://jsbin.com/socibiqudi/edit?html,output

### Steps to reproduce

`		<iron-form id="form">
			<form>
				<vaadin-grid items="[[values]]" >
					<vaadin-grid-column>
						<template class="header">Name</template>
					 <template>
					   <paper-input auto-validate required value="{{item.name}}"></paper-input-input>
					</template>
				   </vaadin-grid-column>
				</vaadin-grid>
			<form>
				
		</iron-form>`

### Browsers Affected

- [ x] Chrome
- [ x] Firefox
- [ ] Safari
- [ ] Edge
- [ ] IE 11
- [ ] iOS Safari
- [ ] Android Chrome

**Comment by jouni** [comment]
I’m still interested in learning why there are those extra cell content elements rendered.

**Comment by tomivirkki** [comment]
The extra cell content elements are slotted into grid's internal sizer rows (2).

**Comment by 200Puls** [comment]
Thanks for the workaround @tomivirkki. This works well for now.
Although I believe this is unexpected behavior to the user and not very intuitive.

**Comment by tomivirkki** [comment]
@jouni The sizer rows are generated with the same function as the other rows so the templates get stamped into them also. Currently, we just [hide the content on sizer rows with `display: none !important`](https://github.com/vaadin/vaadin-grid/blob/master/src/vaadin-grid-styles.html#L286) but might be a worthy optimization to not stamp the templates on sizer rows.

**Comment by jouni** [comment]
@tomivirkki, yeah, sounds like a potential enhancement. Could you write a new issue about that (since you know what to write more precisely)?
